DuckDuckGo/Sync/SyncDiagnosisHelper.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  SyncDiagnosisHelper.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import DDGSync
+import PixelKit
+
+final class SyncDiagnosisHelper {
+    private struct Consts {
+        static let syncManuallyDisabledKey = "com.duckduckgo.app.key.debug.SyncManuallyDisabled"
+        static let syncWasDisabledUnexpectedlyPixelFired = "com.duckduckgo.app.key.debug.SyncWasDisabledUnexpectedlyPixelFired"
+    }
+
+    private let userDefaults = UserDefaults.standard
```

Comment:
instead of direct defaults reading/writing, `@UserDefaultsWrapper`-wrapped variables could be used for better readability and match other places:
```
extension UserDefaultsWrapper.DefaultsKey {
static let syncManuallyDisabledKey = Self(rawValue:  "com.duckduckgo.app.key.debug.SyncManuallyDisabled")
...
}
    @UserDefaultsWrapper(key: . syncManuallyDisabledKey)
    var syncManuallyDisabled: Bool?
    ...
```